pelican/plugins/photos/photos.py

```diff
@@ -2103,85 +2123,109 @@ def replace_inline_images(content, inline_images):
                     prof_attr_name, prof_attr_value.format(i=image)
                 )
 
-        if profile.has_template:
-            content._content = content._content.replace(
-                image_string,
-                profile.render_template(
-                    content=content,
-                    image=image,
-                    match=m,
-                    parsed_attrs=dict(parsed_attrs),
-                ),
-            )
-
-        elif what == "photo":
-            content._content = content._content.replace(
-                image_string,
-                "".join(
-                    (
-                        "<",
-                        m.group("tag"),
-                        m.group("attrs_before"),
-                        m.group("src"),
-                        "=",
-                        m.group("quote"),
-                        "{siteurl}/{filename}".format(
-                            siteurl=pelican_settings["SITEURL"],
-                            filename=image.image.web_filename,
-                        ),
-                        m.group("quote"),
-                        extra_attributes,
-                        m.group("attrs_after"),
-                    )
-                ),
-            )
+        relative_urls = pelican_settings["RELATIVE_URLS"]
+        if relative_urls:
+            siteurl = "."
+        else:
+            siteurl = pelican_settings["SITEURL"]
 
-        elif what == "lightbox" and tag == "img":
-            lightbox_attr_list = [""]
+        if what == "photo":
+            template_file_name = pelican_settings["PHOTO_INLINE_IMAGE_TEMPLATE"]
+        else:
+            template_file_name = pelican_settings["PHOTO_INLINE_LIGHTBOX_TEMPLATE"]
+        try:
+            template = g_generator.get_template(template_file_name)
+        except Exception:
+            template = None
```

Comment:
In this case at least please log a warning or error about the issue, rather than silently bailing.